packages/trpc-playground/src/zod-resolve-types.ts

```diff
@@ -28,12 +31,33 @@ const joinQueries = (functionName: string, queries: Record<string, { inputParser
   return `declare function ${functionName}<QueryName extends ${joinedQueryNames}>(${args.join(',')}): void`
 }
 
-export const zodResolveTypes = (router: AnyRouter) => [
-  joinQueries('query', router._def.queries),
-  joinQueries('mutation', router._def.mutations),
-]
+type DefaultOperationType = { value: string; inputLength: number }
+type QueryDefaultAndType = Record<string, { default: DefaultOperationType; type: string }>
+
+export type GetTypesFromRouterReturn = {
+  queries: QueryDefaultAndType
+  mutations: QueryDefaultAndType
+}
```

Comment:
please extract these types to somewhere shared between the frontend and backend so they don't have to be duplicated

Reply:
Moved it inside `@trpc-playground/types`.